decode.go

```diff
@@ -1266,6 +1266,21 @@ func unquoteBytes(s []byte) (t []byte, ok bool) {
 				b[w] = '\t'
 				r++
 				w++
+			case 'x':
+				if s[r+1] == '1' && (s[r+2] == 'b' || s[r+2] == 'B') {
+					// Interpret as raw escape sequence
+					b[w] = '\x1b'
+					r += 3
+					w++
+					for r < len(s) {
+						if s[r] == '"' {
+							break
+						}
+						b[w] = s[r]
+						r++
+						w++
+					}
```

Comment:
All I meant is that at line 1270 you don't check if `s[r+1]` and `s[r+2]` are within the buffer limits. (It has been checked at line 1232 that `r` is less than `len(s)`, but it hasn't been checked if it is less than `len(s)+2`.) So if the string ends with `\x` or `\x1`, it will probably crash.

Now, looking at this PR a bit more closely, I have more remarks:
- Do we need to parse the entire escape sequence? I think we only need to parse the escape character `\x1b`.
- So this inner `for r < len(s)` loop seems not needed at all. The same is done in the outer loop anyway.
-  Furthermore, why not add generic `\x` support for escaping arbitrary hex bytes, not just the escape byte 0x1b? After all, this is a generic JSON parsing library, it shouldn't need to know about terminal escape sequences or about micro.

Reply:
I've just noticed you've already fixed the out of bounds issue. I'll copy the rest of my comments to the updated version.